### PR TITLE
feat: projects data source

### PR DESCRIPTION
**Pull request by malhussan**

### CODE REVIEW

File: internal/provider/projects_data_source.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	projects, err := d.client.ReadProjects(workspaceIdentifier, paymentMethodIdentifier)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to read projects", err.Error())
+		return
+	}
+
+	var state MeshProjectsSchema
+
+	for _, p := range *projects {
+		project := MeshProject{
```
Comment (henryde) [inline internal/provider/projects_data_source.go:149]: d: `projects` is already a list of `MeshProjects`, why create new ones?

**Review by henryde — CHANGES_REQUESTED** [review]

File: internal/provider/projects_data_source.go (hunk 1 above)
Comment (malhussan) [reply internal/provider/projects_data_source.go]: agreed, no need for it.

File: internal/provider/projects.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package provider
+
+type MeshProjectsSchema struct {
```
Comment (henryde) [inline internal/provider/projects.go:3]: f: This type belongs primarily to the client and should be called `MeshProjects`

File: internal/provider/projects.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+	WorkspaceIdentifier     string        `tfsdk:"workspace_identifier"`
```
Comment (henryde) [inline internal/provider/projects.go:4]: f: should all have json annotations as well

**Review by henryde — CHANGES_REQUESTED** [review]

File: internal/provider/projects.go (hunk 2 above)
Comment (malhussan) [reply internal/provider/projects.go]: I've moved it to client as part of a nested response struct
https://github.com/meshcloud/terraform-provider-meshstack/blob/d77af63254ab0f1405fbd14537fe478f3146a31c/internal/provider/client.go#L307

File: internal/provider/projects_data_source.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		WorkspaceIdentifier     string        `json:"workspaceIdentifier" tfsdk:"workspace_identifier"`
+		PaymentMethodIdentifier *string       `json:"paymentMethodIdentifier" tfsdk:"payment_method_identifier"`
+		Projects                []MeshProject `json:"meshProjects" tfsdk:"projects"`
+	}
+
+	state.Projects = *projects
+	state.WorkspaceIdentifier = workspaceIdentifier
+	state.PaymentMethodIdentifier = paymentMethodIdentifier
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
```
Comment (henryde) [inline internal/provider/projects_data_source.go:155]: s: this whole part could be replaced with 3 `resp.State.SetAttribute(...)` making the nested struct unnecessary.

**Review by henryde — APPROVED** [review]

**Pull request merged**